contracts/l2/L2DAIWormholeBridge.sol

```diff
@@ -85,40 +80,40 @@ contract L2DAIWormholeBridge is OVM_CrossDomainEnabled {
     address operator
   ) external {
     WormholeGUID memory wormhole = WormholeGUID({
-      sourceDomain: sourceDomain,
+      sourceDomain: domain,
       targetDomain: targetDomain,
       receiver: receiver,
       operator: operator,
       amount: amount,
-      nonce: uint64(OVM_L2CrossDomainMessenger(address(getCrossDomainMessenger())).messageNonce()), // gas optimization, we don't need to maintain our own nonce
-      timestamp: uint64(block.timestamp)
+      nonce: uint80(OVM_L2CrossDomainMessenger(address(getCrossDomainMessenger())).messageNonce()), // gas optimization, we don't need to maintain our own nonce
+      timestamp: uint48(block.timestamp)
     });
 
     batchedDaiToFlush[targetDomain] += amount;
     Mintable(l2Token).burn(msg.sender, amount);
 
-    uint32 l1Gas = 20000; // @todo: does it matter? messages need to be relied manually anyway
     bytes memory message = abi.encodeWithSelector(
       L1DAIWormholeBridge.finalizeRegisterWormhole.selector,
       wormhole
     );
-    sendCrossDomainMessage(l1DAITokenBridge, l1Gas, message);
+    sendCrossDomainMessage(l1DAIWormholeBridge, 0, message);
 
     emit WormholeInitialized(wormhole);
   }
 
   function flush(bytes32 targetDomain) external {
     uint256 daiToFlush = batchedDaiToFlush[targetDomain];
+    require(daiToFlush > 0, "L2DAIWormholeBridge/no-dai-to-flush");
```

Comment:
Thinking more on this, I think we should allow 0 DAI flushes. I'm thinking likely we will need to keep track of the debt, settlements on the target domain over time to do bad debt settlement. Using the flush / settle functions as the trackers seems likely to me. Having a daily heartbeat regardless of if there is DAI to push is probably a good idea.

Reply:
Yes, that makes sense